students/uss_tomasz/lesson_03_functions/calculator.py

```diff
@@ -0,0 +1,64 @@
+def display_options():
+    """This function is needed in order to display which are possible inputs for user.
+    """
+    print("a - add")
+    print("s - subtract")
+    print("m - multiply")
+    print("d - divide")
+    print("p - power")
+    print("h,? - help")
+    print("q - QUIT")
+
+
+def collect_variables():
+    """Function is asking user to write two integers.
+    Returns:
+        Two variables. They are needed to calculate result, for example adding
+    """
+    var_1 = int(input("Input 1st operand:"))
+    var_2 = int(input("Input 2nd operand:"))
+    return var_1, var_2
+
+
+def execute_command(command_name, function_type):
```

Comment:
very good and clean solution
one remark - `function_type` suggest that only some kind of "type" is passed here, whilst we're injecting the function itself. `result_function` or `execute_operation` would be more meaningful in this case, but it's a minor, the rest is really perfect

what you've done here is known as strategy pattern - slide 62: https://fsharpforfunandprofit.com/fppatterns/